torch/_dynamo/variables/torch.py

```diff
@@ -782,26 +782,32 @@ def speculate_subgraph(
     tx, f, sub_args, graph_checkpoint, checkpoint, *, always_restore=False
 ):
     from . import AutogradFunctionContextVariable, ConstantVariable, TensorVariable
+    from .builder import wrap_fx_proxy
 
     try:
         with tx.output.new_subtracer() as tracer:
             args = []
             # One argument to graph per sub_args
             for a in sub_args:
                 if a is None:
-                    a = ConstantVariable(None)
+                    new_arg = ConstantVariable(None)
                 assert not isinstance(
                     a, torch.Tensor
                 ), "Tensors should already be tracked?"
                 if isinstance(a, ConstantVariable):
                     proxy = tracer.create_graph_input("const")
+                    new_arg = a
                 elif isinstance(a, (TensorVariable, AutogradFunctionContextVariable)):
-                    tracer.create_graph_input(a.as_proxy().node.name)
+                    new_proxy = tracer.create_graph_input(a.as_proxy().node.name)
+                    example_value = a.as_proxy().node.meta["example_value"]
+                    new_arg = wrap_fx_proxy(
+                        tx=tx, proxy=new_proxy, example_value=example_value
+                    )
```

Comment:
This looks like it's causing one of the autograd.Function test cases to fail? `ReproTests.test_hf_xsoftmax_training`